astropy/io/fits/tests/test_connect.py

```diff
@@ -267,6 +267,24 @@ def test_oned_single_element(self, tmpdir):
         assert read['x'].shape == (2, 1)
         assert len(read['x'][0]) == 1
 
+    def test_write_append(self, tmpdir):
+        filename = str(tmpdir.join('test_write_append.fits'))
+        t = Table(self.data)
+        t.write(filename, append=True)
+        t.write(filename, append=True)
+        with fits.open(filename) as hdu_list:
+            assert len(hdu_list) == 3
```

Comment:
Should we also check the content? Although maybe that is pointless here given this test is appending the same stuff over and over again. Should we modify the test to append different data to better mimic real life usage?

Reply:
Yes, seems useful to check that the HDUs are correct.

Reply:
I have updated the tests to check the content.